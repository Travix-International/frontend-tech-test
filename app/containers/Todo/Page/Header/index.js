import React from 'react';
import { FormattedMessage } from 'react-intl';

import messages from '../../messages';
import Wrapper from './Wrapper';
import Label from './Label';

const TodoHeader = () => {
  return (
    <Wrapper>
      <Label>
        <FormattedMessage {...messages.title} />
      </Label>
      <Label>
        <FormattedMessage {...messages.description} />
      </Label>
      <Label></Label>
    </Wrapper>
  );
};

export default TodoHeader;
