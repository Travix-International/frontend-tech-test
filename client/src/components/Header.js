import React from 'react';
import '../styles/layout/header.sass';

const Header = () => (
  <header>
    <h1> To-Do list</h1>
  </header>
);

export default Header;
