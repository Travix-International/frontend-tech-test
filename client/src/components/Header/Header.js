import React from 'react'
import PropTypes from 'prop-types'

import styles from './Header.scss'

const Header = props => {
  const { counter } = props
  return (
    <div className={styles.Header}>
      <h1>Travix ToDo list</h1>
      <div className={styles.Counter}>
        <p>{`Tasks To Do: ${counter}`}</p>
      </div>
    </div>
  )
}

Header.propTypes = {
  counter: PropTypes.number.isRequired,
}

export default Header
