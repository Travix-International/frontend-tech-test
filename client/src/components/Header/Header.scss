header {
  display: grid;
  height: 100px;
  grid-auto-flow: column;
  align-items: center;
  justify-content: space-around;
  box-shadow: 0 2px 6px 0px #cecece;
  background-image: linear-gradient(135deg, #f761a1 10%, #8c1bab 100%);

  h1 {
    color: #fff;
  }
}
