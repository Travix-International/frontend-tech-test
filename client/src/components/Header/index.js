import React from 'react';
import {Link,withRouter} from 'react-router-dom';
import {connect} from 'react-redux';
import AppBar from '@material-ui/core/AppBar';
import Avatar from '@material-ui/core/Avatar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import {Dropdown, DropdownMenu, DropdownToggle} from 'reactstrap';
import {
    BELOW_THE_HEADER,
    COLLAPSED_DRAWER,
    FIXED_DRAWER,
    HORIZONTAL_NAVIGATION,
    INSIDE_THE_HEADER
} from 'constants/ActionTypes';
import {switchLanguage, toggleCollapsedNav} from 'actions/Setting';
import LanguageSwitcher from 'components/LanguageSwitcher/index';
import Menu from 'components/TopNav/Menu';

class Header extends React.Component {


  onLangSwitcherSelect = (event) => {
    this.setState({
      langSwitcher: !this.state.langSwitcher, anchorEl: event.currentTarget
    })
  };

  onAppsSelect = () => {
    this.setState({
      apps: !this.state.apps
    })
  };
  onUserInfoSelect = () => {
    this.setState({
      userInfo: !this.state.userInfo
    })
  };
  handleRequestClose = () => {
    this.setState({
      langSwitcher: false,
      userInfo: false,
      apps: false
    });
  };
  onToggleCollapsedNav = (e) => {
    const val = !this.props.navCollapsed;
    this.props.toggleCollapsedNav(val);
  };

  constructor() {
    super();
    this.state = {
      anchorEl: undefined,
      userInfo: false,
      langSwitcher: false
    }
  }

 

  render() {
    const {drawerType, locale, navigationStyle, horizontalNavPosition} = this.props;
    const drawerStyle = drawerType.includes(FIXED_DRAWER) ? 'd-block d-xl-none' : drawerType.includes(COLLAPSED_DRAWER) ? 'd-block' : 'd-none';

    return (
      <AppBar
        className={`app-main-header ${(navigationStyle === HORIZONTAL_NAVIGATION && horizontalNavPosition === BELOW_THE_HEADER) ? 'app-main-header-top' : ''}`}>
        <Toolbar className="app-toolbar" disableGutters={false}>
          {navigationStyle === HORIZONTAL_NAVIGATION ?
            <div className="d-block d-md-none pointer mr-3" onClick={this.onToggleCollapsedNav}>
                            <span className="jr-menu-icon">
                              <span className="menu-icon"/>
                            </span>
            </div>
            :
            <IconButton className={`jr-menu-icon mr-3 ${drawerStyle}`} aria-label="Menu"
                        onClick={this.onToggleCollapsedNav}>
              <span className="menu-icon"/>
            </IconButton>
          }

          <Link className="app-logo mr-2 d-none d-sm-block" to="/">
            <img src={require("assets/images/logo.png")} alt="Travix" title="Travix"/>
          </Link>


          {(navigationStyle === HORIZONTAL_NAVIGATION && horizontalNavPosition === INSIDE_THE_HEADER) &&
          <Menu/>}

          <ul className="header-notifications list-inline ml-auto">
            <li className="list-inline-item">
              <Dropdown
                className="quick-menu"
                isOpen={this.state.langSwitcher}
                toggle={this.onLangSwitcherSelect.bind(this)}>

                <DropdownToggle
                  className="d-inline-block"
                  tag="span"
                  data-toggle="dropdown">
                  <IconButton className="icon-btn">
                    <i className={`flag flag-24 flag-${locale.icon}`}/>
                  </IconButton>
                </DropdownToggle>

                <DropdownMenu right className="w-50">
                  <LanguageSwitcher switchLanguage={this.props.switchLanguage}
                                    handleRequestClose={this.handleRequestClose}/>
                </DropdownMenu>
              </Dropdown>


            </li>


            {navigationStyle === HORIZONTAL_NAVIGATION &&
            <li className="list-inline-item user-nav">
              <Dropdown
                className="quick-menu"
                isOpen={this.state.userInfo}
                toggle={this.onUserInfoSelect.bind(this)}>

                <DropdownToggle
                  className="d-inline-block"
                  tag="span"
                  data-toggle="dropdown">
                  <IconButton className="icon-btn size-30">
                    <Avatar
                      alt='...'
                      src={'https://via.placeholder.com/150x150'}
                      className="size-30"
                    />
                  </IconButton>
                </DropdownToggle>

              </Dropdown>
            </li>}
          </ul>
          
          <div className="ellipse-shape"></div>
        </Toolbar>
      </AppBar>
    );
  }

}


const mapStateToProps = ({settings}) => {
  const {drawerType, locale, navigationStyle, horizontalNavPosition} = settings;
  return {drawerType, locale, navigationStyle, horizontalNavPosition}
};

export default withRouter(connect(mapStateToProps, {toggleCollapsedNav, switchLanguage})(Header));