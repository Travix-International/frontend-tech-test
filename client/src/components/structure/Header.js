import React from 'react';

export const Header = () => (
  <header className="main-header">
    <h1 className="logo">
      <figure>Travix <span>TODO</span></figure>
    </h1>
  </header>
);

export default Header;
