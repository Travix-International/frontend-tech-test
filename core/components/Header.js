import React, { Component } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch, faSort } from '@fortawesome/free-solid-svg-icons';

class Header extends Component {
  render() {
    return (
      <div id="header">



      </div>
    );
  }
}


export default Header;
