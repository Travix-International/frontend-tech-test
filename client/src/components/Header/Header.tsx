import React from "react";

import "./Header.scss";

const Header: React.FC = () => {
  return (
    <header>
      <h1>ToDo</h1>
    </header>
  );
};

export default Header;
