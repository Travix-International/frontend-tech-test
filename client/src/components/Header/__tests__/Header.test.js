import React from 'react'
import {shallow} from 'enzyme'
import Header from '../Header'

describe('App should render properly', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Header />)
  })
  it('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot('Header_snapshot_1')
  })

  it('Should show "Tasks To Do: 3" if passed counter = 3', () => {
    console.log(wrapper.debug())
    wrapper.setProps({counter: 3})
    console.log(wrapper.debug())
    expect(
      wrapper
        .find('div.Header')
        .render()
        .text()
    ).to.equal('Tasks To Do: 3')
  })
})
