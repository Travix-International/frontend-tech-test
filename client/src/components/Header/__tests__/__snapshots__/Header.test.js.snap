// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly Should match snapshot: Header_snapshot_1 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Travix ToDo list
        </h1>,
        <div
          className="Counter"
        >
          <p>
            Tasks To Do: 
          </p>
        </div>,
      ],
      "className": "Header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Travix ToDo list",
        },
        "ref": null,
        "rendered": "Travix ToDo list",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Tasks To Do: 
          </p>,
          "className": "Counter",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Tasks To Do: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Tasks To Do: ",
            undefined,
          ],
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Travix ToDo list
          </h1>,
          <div
            className="Counter"
          >
            <p>
              Tasks To Do: 
            </p>
          </div>,
        ],
        "className": "Header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Travix ToDo list",
          },
          "ref": null,
          "rendered": "Travix ToDo list",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Tasks To Do: 
            </p>,
            "className": "Counter",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Tasks To Do: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Tasks To Do: ",
              undefined,
            ],
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
